code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -171,44 +171,94 @@
 				H.update_hair()
 
 		if("Hair Style")
-			//facial hair
+
+			//Facial hair
+
 			if(H.gender == MALE)
-				var/new_style = input(owner, "Select a facial hair style.", "Facial Hair Alterations") as null|anything in GLOB.facial_hairstyles_list
+				var/new_style = input(owner, "Select a facial hair style.", "Facial Hair Alteration") as null|anything in GLOB.facial_hairstyles_list
 				if(new_style)
 					H.facial_hairstyle = new_style
 			else
-				H.facial_hairstyle = "Shaved" //Female characters can't have beards
-			//normal hair
-			var/new_style = input(owner, "Select your hair style.", "Hair Style Alterations") as null|anything in GLOB.hairstyles_list
+				H.facial_hairstyle = "Shaved"
+				H.update_hair()
+
+			//Normal hair
+
+			var/new_style = input(owner, "Select your hair style.", "Hair Style Alteration") as null|anything in GLOB.hairstyles_list
 			if(new_style)
 				H.hairstyle = new_style
 				H.update_hair()
-				//Ears
-		if("Ears")
-			var/selected_ears = input(owner, "Select your desired ears.", "Ear Alteration") in list("None", "Cat") //easily expandable in case we ever have more ears
-			if(selected_ears)
-				switch(selected_ears)
-					if("None")
-						H.dna.features["ears"] = "None"
-						H.dna.species.mutant_bodyparts -= "ears"
-						H.update_body()
-					if("Cat")
-						H.dna.species.mutant_bodyparts |= "ears"
-						H.dna.features["ears"] = "Slimecat"
-						H.update_body()
-				//Tails
-		if("Tail")
-			var/selected_tail = input(owner, "Select your desired tail.", "Tail Alteration") in list("None", "Cat") //lizard tails and/or horns to follow eventually
-			if(selected_tail)
-				switch(selected_tail)
-					if("None")
-						H.dna.features["tail_human"] = "None"
-						H.dna.species.mutant_bodyparts -= "tail_human"
-						H.update_body()
-					if("Cat")
-						H.dna.species.mutant_bodyparts |= "tail_human"
-						H.dna.features["tail_human"] = "Slimecat"
-						H.update_body()
+
+				//Mutant Bodyparts
+
+		if("Mutant Bodyparts")
+			var/list/part_list = list("Ears", "Tails", "Frills", "Spines", "Wings") //Selection, if you want more stuff added, do it yourself
+			var/selected_part = input(owner, "Which bodypart do you want to alter?", "Mutant Bodypart Alteration") as null|anything in part_list + "Cancel"
+			if(selected_part)
+				if(selected_part == "Cancel")
+					return
+				switch(selected_part)
+					if("Ears")
+						handle_sprites(lowertext(selected_part), H, GLOB.ears_list)
+					if("Tails") //Tails are special and need special treatment because they're a combination of 2 lists.
+						handle_tails(H)
+					if("Frills")
+						handle_sprites(lowertext(selected_part), H, GLOB.frills_list)
+					if("Spines")
+						handle_sprites(lowertext(selected_part), H, GLOB.spines_list)
+					if("Wings") //These don't look too good
+						handle_sprites(lowertext(selected_part), H, GLOB.wings_list)
+	return
```

Comment:
Just dump an `update_body(H)` before the return and remove all the other update_body() calls